File: src/main/java/org/eln/eln3/sim/nbt/NbtFurnaceProcess.java
package org.eln.eln3.sim.nbt;


import net.minecraft.nbt.CompoundTag;
import org.eln.eln3.sim.FurnaceProcess;
import org.eln.eln3.sim.ThermalLoad;
import org.jetbrains.annotations.NotNull;

public class NbtFurnaceProcess extends FurnaceProcess implements TagSerializable {

    String name;

    public NbtFurnaceProcess(String name, ThermalLoad load) {
        super(load);
        this.name = name;
    }

    @Override
    public void loadAdditionalData(@NotNull CompoundTag nbt, @NotNull String str) {
        combustibleEnergy = nbt.getFloat(str + name + "Q");
        setGain(nbt.getDouble(str + name + "gain"));
    }

    @Override
    public void saveAdditionalData(@NotNull CompoundTag nbt, @NotNull String str) {
        nbt.putFloat(str + name + "Q", (float) combustibleEnergy);
        nbt.putDouble(str + name + "gain", getGain());
    }
}

File: src/main/kotlin/org/eln/eln3/sim/nbt/NbtFurnaceProcess.kt
package org.eln.eln3.sim.nbt

import net.minecraft.nbt.CompoundTag
import org.eln.eln3.sim.process.thermal.FurnaceProcess
import org.eln.eln3.sim.thermal.ThermalLoad

class NbtFurnaceProcess(var name: String?, load: ThermalLoad) : FurnaceProcess(load), TagSerializable {
    override fun loadAdditionalData(nbt: CompoundTag, str: String) {
        combustibleEnergy = nbt.getFloat(str + name + "Q").toDouble()
        setGain(nbt.getDouble(str + name + "gain"))
    }

    override fun saveAdditionalData(nbt: CompoundTag, str: String) {
        nbt.putFloat(str + name + "Q", combustibleEnergy.toFloat())
        nbt.putDouble(str + name + "gain", getGain())
    }
}